app/services/course_groups/leadership.rb

```diff
@@ -10,10 +10,8 @@ class Leadership
     delegate :schedules, to: :course_group
 
     def schedule
-      if autumn_schedule_2022?(schedule_date)
-        schedules.find_by!(cohort:, identifier: "npq-leadership-autumn")
-      elsif autumn_schedule_2024?(schedule_date)
-        schedules.find_by!(cohort:, identifier: "npq-leadership-autumn")
+      if autumn_schedule_2024?(schedule_date)
+        schedules.find_by(cohort:, identifier: "npq-leadership-autumn") || schedules.find_by!(cohort:, identifier: "npq-leadership-spring")
```

Comment:
```suggestion
        if cohort.start_year == 2025
          schedules.find_by!(cohort:, identifier: "npq-leadership-spring")
        elsif cohort.start_year == 2024
          schedules.find_by(cohort:, identifier: "npq-leadership-autumn")
        end
```
an alternative implementation is this.

Reply:
This is clearer still actually and makes it obvious that 2024 and 2025 are different.